core/src/wsv.rs

```diff
@@ -319,6 +322,49 @@ impl<W: WorldTrait> WorldStateView<W> {
             .map(|block_entry| block_entry.value().clone())
     }
 
+    /// Get `World` and pass it to closure to modify it
+    ///
+    /// Produces events after `f` finishes. Events are emitted from the lowest to the highest
+    ///
+    /// # Errors
+    /// Fails if `f` fails
+    pub fn modify_world(
+        &self,
+        f: impl FnOnce(&World) -> Result<WorldEvent, Error>,
+    ) -> Result<(), Error> {
+        let mut events: SmallVec<[DataEvent; 3]> = SmallVec(smallvec::smallvec![]);
+        let event = f(&self.world)?;
+
+        match &event {
```

Comment:
can you match by value here and avoid clones?

Reply:
Yeah, I little bit less clones now